common/flatpak-transaction.c

```diff
@@ -3158,6 +3158,11 @@ resolve_ops (FlatpakTransaction *self,
           /* We resolve to the deployed metadata, because we need it to uninstall related ops */
 
           metadata_bytes = load_deployed_metadata (self, op->ref, &checksum, NULL);
+          if (metadata_bytes == NULL)
```

Comment:
I think you should check `checksum` here. That's what the assertion is on, `mark_op_resolved` is tolerant of missing metadata, and the metadata isn't actually used for uninstall operations.

Reply:
While it is true that its the checksum that is important, the return value is the signal for failure, so seems weird to not check that. And I don't think this will ever fail with no metadata, but a checksum.